### PR TITLE
(2061) Show report breadcrumb scheme when drilling down from a report

### PR DESCRIPTION
**Pull request by pezholio**
This PR adds a breadcrumb when drilling down to a report from the report listing page, and also adds a `BreadcrumbContext` class that allows users to drill down to an Activity from a report whilst still keeping the breadcrumbs within the context of the report.

I've also made a not insignificant refactor to gather all the activity-related controllers and put them into one controller, which makes some of this work easier, as well as tidying up the codebase and making the location of the controller code for activities much easier to find.

# Screenshots

image

image

image

### CODE REVIEW

**Comment by pezholio** [comment]
[Show "Report" breadcrumb scheme when drilling down from a report](https://trello.com/c/v2XTTo0t/2061-show-report-breadcrumb-scheme-when-drilling-down-from-a-report)

File: app/services/activity/tab.rb [hunk 1]
```diff
@@ -0,0 +1,80 @@
+class Activity
+  class Tab
+    include Pundit
```
Comment (edavey) [inline app/services/activity/tab.rb:3]: I'm a bit uneasy with such a complex and clever abstraction of the activity tabs:

- use of `send`
- use of `instance_variable_get`
- dynamically generated routes

BUT: you explain that this is a reaction to real pain experienced maintaining these views, so yes, let's give it a go!

**Review by edavey — APPROVED** [review]
👍 really nice work! 

I can't find anything to tweak. 

I do have some reservations about the slightly magical `Activity::Tab`, but very happy to see how it goes with less controllers.

**Pull request merged**